# Introduction To Probability
## Challenge 1

A and B are events of a probability space with $(\omega, \sigma, P)$ such that $P(A) = 0.3$, $P(B) = 0.6$ and $P(A \cap B) = 0.1$

Which of the following statements are false?
* $P(A \cup B) = 0.6$
* $P(A \cap B^{C}) = 0.2$
* $P(A \cap (B \cup B^{C})) = 0.4$
* $P(A^{C} \cap B^{C}) = 0.3$
* $P((A \cap B)^{C}) = 0.9$

## Answer 1
wrong:     $P(A \cup B) = 0.6$           
correct:   $P(A \cup B) = 0.8$   

wrong:     $P(A \cap (B \cup B^{C})) = 0.4$         
correct:   $P(A \cap (B \cup B^{C})) = 0.3$

wrong:     $P(A^{C} \cap B^{C}) = 0.3$                          
correct:   $P(A^{C} \cap B^{C}) = 0.2$

## Challenge 2
There is a box with 10 white balls, 12 red balls and 8 black balls. Calculate the probability of:
* Taking a white ball out.
* Taking a white ball out after taking a black ball out.
* Taking a red ball out after taking a black and a red ball out.
* Taking a red ball out after taking a black and a red ball out with reposition.

**Hint**: Reposition means putting back the ball into the box after taking it out.

## Answer 2
Taking a white ball out (assuming we take 3 balls):                                                                  
p = (10/30)*(9/29+12/29+8/29) + (12/30)*(10/29) + (8/30)*(10/29) = 0.563218                  
p = (10/30)* (20/29) + (20/30)*(10/29) + (10/30)*(9/29) = 0.563218

Taking a white ball out (assuming we take 1 ball):   
p = (10/30)= 0.3333

Taking a white ball out after taking a black ball out:                                               
p = (8/30)*(10/29) = 0.09195

Taking a red ball out after taking a black and a red ball out:                                    
p = ((12/30)*(8/29) + (8/30)*(12/29)) * (11/28) = 0.0867

Taking a red ball out after taking a black and a red ball out with reposition:                     
p = ((12/30)*(8/30) + (8/30)*(12/30)) * (12/30) = 0.085333

## Challenge 3

You are planning to go on a picnic today but the morning is cloudy. You hate rain so you don't know whether to go out or stay home! To help you make a decision, you gather the following data about rainy days:

* 50% of all rainy days start off cloudy!
* Cloudy mornings are common. About 40% of days start cloudy. 
* This month is usually dry so only 3 of 30 days (10%) tend to be rainy. 

What is the chance of rain during the day?

## Answer 3
$P(rain \mid cloudy)$ = $\frac{P(rain \cap cloudy)}{P(cloudy)}$ = $\frac{P(cloudy \mid rain)*P(rain)}{P(cloudy)}$ = $\frac{0,5*0,1}{0,4}$ = 0,125

## Challenge 4

One thousand people were asked through a telephone survey whether they thought more street lighting is needed at night or not.

Out of the 480 men that answered the survey, 324 said yes and 156 said no. On the other hand, out of the 520 women that answered, 351 said yes and 169 said no. 

We wonder if men and women have a different opinions about the street lighting matter. Is gender relevant or irrelevant to the question?

Consider the following events:
- The answer is yes, so the person that answered thinks that more street lighting is needed.
- The person who answered is a man.

We want to know if these events are independent, that is, if the fact of wanting more light depends on whether one is male or female. Are these events independent or not?

**Hint**: To clearly compare the answers by gender, it is best to place the data in a table.

In [2]:
import pandas as pd

In [4]:
# table
pd.DataFrame({"men":{"yes":324,"no":156,"total":480},"women":{"yes":351,"no":169,"total":520}, "total":{"yes":675,"no":325,"total":1000}})

men  women  total
yes    324    351    675
no     156    169    325
total  480    520   1000

## Answer 4
Are independent if:                                                  
$P(yes \cap man)$ = $P(yes)*P(man)$     

$P(yes \cap man)$ = 324/1000 = 0,324                                  
$P(yes)*P(man)$ = (675/1000)*(480/1000) = 0,324     

Ansewrs: Are independent